classes/security/authorization/internal/UserAccessibleWorkflowStageRequiredPolicy.inc.php

```diff
@@ -20,13 +20,21 @@ class UserAccessibleWorkflowStageRequiredPolicy extends AuthorizationPolicy {
 	/** @var PKPRequest */
 	var $_request;
 
+	/** @var string Workflow type. One of WORKFLOW_TYPE_... **/
+	var $_workflowType;
+
 	/**
 	 * Constructor
 	 * @param $request PKPRequest
+	 * @param $workflowType string Which workflow the stage access must be granted
+	 *  for. One of WORKFLOW_TYPE_*.
 	 */
-	function __construct($request) {
+	function __construct($request, $workflowType = null) {
 		parent::__construct('user.authorization.accessibleWorkflowStage');
 		$this->_request = $request;
+		if (!is_null($workflowType)) {
```

Comment:
I don't think this conditional does anything -- `$this->workflowType` will be null anyway if it's not initialized here.